pam/pam_adsys.c

```diff
@@ -233,6 +234,11 @@ static char *get_default_sss_domain(pam_handle_t *pamh) {
     }
     fclose(f);
 
+    if (domain == NULL) {
+        free(buf);
+        return NULL;
+    }
+
```

Comment:
If `domain` is NULL in the end we need to return earlier, otherwise it can cause problems in `strdup()`

Reply:
Unrelated to your PR, but could `*++domain` evaluate to `\t`, for instance?

Reply:
It could, yeah. Why?

Reply:
This part of the code

```c
while (*++domain == ' ') continue;
            char *newline = strchr(domain, '\n');
            if (newline != NULL) {
                *newline = '\0';
            }
            break;
```

allows for `domain`  to contain `\t` and other space-like characters. Imagine a line in sssd.conf like:

```
default_domain_suffix\t=\tmydomain.com
```

It seems to allow for the above. Could that be a problem?

Reply:
Indeed, it seems like an issue. We'll have to test and see how sssd proceeds in these kinds of situation, but it they account for it, we must do the same as well.
Thanks for spotting it!